File: aat-lib/src/main/java/ch/bailu/aat_lib/view/graph/Segment.java
package ch.bailu.aat_lib.view.graph;

public class Segment {
    private int firstPoint = -1;
    private int lastPoint = -1;

    public void setLimit(int first, int last) {
        firstPoint = first;
        lastPoint = last;
    }

    public boolean hasLimit() {
        return firstPoint > -1 && lastPoint > firstPoint;
    }

    public boolean isAfter(int index) {
        return index > lastPoint;
    }

    public boolean isBefore(int index) {
        return index < firstPoint;
    }

    public boolean isNotAfter(int index) {
        return index <= lastPoint;
    }

    public boolean isInside(int index) {
        return index >= firstPoint && index <= lastPoint;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/view/graph/Segment.kt
package ch.bailu.aat_lib.view.graph


class Segment {
    private var start = -1
    private var end = -1

    fun setLimit(start: Int, end: Int) {
        this.start = start
        this.end = end
    }

    fun isValid(): Boolean {
        return start > -1 && end > start
    }

    fun isAfter(index: Int): Boolean {
        return index > end
    }

    fun isBefore(index: Int): Boolean {
        return index < start
    }

    fun isInside(index: Int): Boolean {
        return index >= start && index <= end
    }
}
